lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1330,6 +1374,77 @@ def _parse_log_config(self):
         except ValueError as exc:
             self.fail('Error parsing logging options - %s' % (exc))
 
+    def _parse_healthcheck(self):
+        '''
+        Return dictionary of healthcheck parameters
+        '''
+        if (not self.healthcheck) or (not self.healthcheck.get('test')):
+            return None, None
+
+        result = dict()
+
+        # all the supported healthecheck parameters
+        options = dict(
+            test='test',
+            interval='interval',
+            timeout='timeout',
+            start_period='start_period',
+            retries='retries'
+        )
+
+        duration_options = ['interval', 'timeout', 'start_period']
+
+        for (key, value) in options.items():
+            if value in self.healthcheck:
+                if value in duration_options:
+                    time = self._convert_duration_to_nanosecond(self.healthcheck.get(value))
```

Comment:
Should we pass it into the `result` if `time` is zero?

Reply:
Hmm, I think so. If the user specified zero, it probably has a good reason (or should fail if docker doesn't like it).

Reply:
In that case, we'd also better drop it, otherwise we'll get idempotency problems...